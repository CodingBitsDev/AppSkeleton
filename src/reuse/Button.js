// @flow
import React, {useState} from "react";
import PropTypes from 'prop-types';
import {Dimensions, ActivityIndicator, Platform, Text, TouchableOpacity, TouchableNativeFeedback, TouchableHighlight, View, } from "react-native";
import { LinearGradient } from 'expo-linear-gradient';

import getTheme from "theme/index.js";
import { getContrastColor } from "../helperFunctions/colorHelpers.js";
import { makeStyle, extractPaddingStyles } from "../helperFunctions/styleHelper.js";

function Button({ style, start, end, locations, ...props }){
  let [contentSize, setContentSize] = useState(null);
  let [containerSize, setContainerSize] = useState(null);
  let [hover, setHover] = useState(false);

  let { colors, styles } = getTheme();

  let extraStyles = {}
  let btnColor = colors.default;
  if (props.disabled){
    btnColor = colors.disabled;
  } else if (props.primary){
    btnColor = colors.primary;
  } else if (props.secondary){
    btnColor = colors.secondary;
  } else if (props.success){
    btnColor = colors.success;
  } else if (props.danger){
    btnColor = colors.danger;
  } else if (props.warning){
    btnColor = colors.warning;
  } else if (props.info){
    btnColor = colors.info;
  }
  btnColor = props.color || (style ? makeStyle(style).backgroundColor : null) || btnColor;
  btnColor = props.disabled ? props.disabledColor || colors.disabled : btnColor;

  let textColor = getContrastColor(btnColor);
  textColor = props.textColor || textColor;
   
  let [ paddingStyle, containerStyle ] = extractPaddingStyles(props.containerStyle);
  let textStyle = makeStyle(props.textStyle);

  let maxScaleIncrease = 5;
  let scaleX = containerSize && containerSize.height > maxScaleIncrease * 2 ? 1 + maxScaleIncrease/containerSize.height : 1.1;
  let scaleY = containerSize && containerSize.width > maxScaleIncrease * 2 ? 1 + maxScaleIncrease/containerSize.width : 1.1;
  let scale = Math.min( scaleX, scaleY );
  let btnStyle = {
    container: {
      minHeight: 30,
      backgroundColor: "transparent",
      alignItems: "center",
      justifyContent: "center",
      overflow: "hidden",
      ...(props.outlined && !props.transparent ? {...styles.border, borderColor: btnColor } : {} ),
      ...(props.full ? styles.roundConers.none : props.rounded ? styles.roundConers.round : styles.roundConers.tiny),
      ...(props.full ? { borderRightWidth: 0, borderLeftWidth:0, width: "100%"} : {} ),
      ...(props.block ? { width: "95%",} : {} ),
      ...(props.noShadow || props.transparent || props.outlined ? {} : styles.shadow ),
      ...( containerStyle || {} ),
      ...(hover & !props.noHover && !props.disabled ? { transform: [ { scaleX: scale }, { scaleY: scale } ] } : {}),
    },
    text: {
      color: props.transparent || props.outlined ? btnColor : textColor,
      fontSize: props.fontSize || 16,
      ...( textStyle || {} ),
    },
    contentStyle:{
      padding: props.transparent ? 0 : 10,
      alignSelf: "stretch",
      flexGrow: 1,
      justifyContent: "center",
      alignItems: "center" 
    },
  };

  let gradientColors = props.disabled ? ( props.disabledColors || [btnColor, btnColor] ) : ( props.colors || [btnColor, btnColor] );
  gradientColors = props.transparent || props.outlined ? [ "transparent", "transparent" ] : gradientColors;

  return (
      <Touchable
        onLayout={ event => { setContainerSize({width: event.nativeEvent.layout.width, height: event.nativeEvent.layout.height}) }}
        style={btnStyle.container} {...props}>
        <LinearGradient 
          onMouseEnter={() => setHover( true )}
          onMouseLeave={() => setHover( false )}
          locations={locations}
          start={start}
          end={end}
          colors={gradientColors}
          style={[ btnStyle.contentStyle ,paddingStyle]}
          onLayout={ event => { setContentSize({width: event.nativeEvent.layout.width, height: event.nativeEvent.layout.height}) }}
        >
        <React.Fragment>
          {props.iconLeft && props.iconLeft}
          <Text style={btnStyle.text}>{ props.title }</Text>
          {props.iconRight && props.iconRight}
        </React.Fragment>
        </LinearGradient>
      </Touchable>
  );
}


function Touchable(props){
  return ( Platform.OS === "ios" ? (
    <TouchableOpacity { ...props} />
  ) : (
    <TouchableOpacity {...props} />
  ));
}

function removeUndefinedFromObject({...obj}){
  Object.keys(obj).forEach(key => {
    if (obj[key] === undefined) {
      delete obj[key];
    }
  });
  return obj;
}

Button.propTypes = {
  //ColorProps
  disabled: PropTypes.bool,
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  success: PropTypes.bool,
  danger: PropTypes.bool,
  warning: PropTypes.bool,
  info: PropTypes.bool, 
  color: PropTypes.string,
  disabledColor: PropTypes.string,
  textColor: PropTypes.string,
  //ShapeProps
  full: PropTypes.bool,
  block: PropTypes.bool,
  rounded: PropTypes.bool,
  outlined: PropTypes.bool,
  transparent: PropTypes.bool,
  //StyleProps
  fontSize: PropTypes.number,
  noShadow: PropTypes.bool,
  //ExtraStyles
  containerStyle: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  textStyle: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  iconLeft: PropTypes.oneOfType([PropTypes.element, PropTypes.object]),
  iconRight: PropTypes.oneOfType([PropTypes.element, PropTypes.object]),
  colors: PropTypes.array,
  disabledColors: PropTypes.array,
  start: PropTypes.array,
  end: PropTypes.array,
  locations: PropTypes.array,
  //Interaction Props
  noHover: PropTypes.bool,
  //StandardProps
  style: PropTypes.object,
  //DataProps
  title: PropTypes.title,
};

export default Button
