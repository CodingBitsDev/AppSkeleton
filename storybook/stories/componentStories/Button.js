import React from 'react';
import { Text, View } from 'react-native';

import { storiesOf } from '@storybook/react-native';

//Components
import Button from "src/reuse/Button.js";

//Decorators
const Center = ({children}) => { return (<View style={{ flex:1, justifyContent: "center", alignItems: "center" }}>{children}</View>) }

const createButtonStory = ( storyName, button ) => {
  storiesOf('Button', module).add(storyName, () => <Center>{button}</Center>);
}

//Start
storiesOf( "Button", module).add("All", () => (
  <View style={{ flex:1, justifyContent: "space-around", alignItems: "center"  }}>
    <Button title="Regular"/>
    <Button rounded title="Rounded"/>
    <Button full title="Full"/>
    <Button block title="Block"/>
    <Button disabled title="Diabled"/>
    <Button primary title="Primary"/>
    <Button secondary title="Secondary"/>
    <Button info title="Info"/>
    <Button success title="Success"/>
    <Button warning title="Warning"/>
    <Button danger title="Danger"/>
    <Button containerStyle={{ width: 300 }} title="Regular, width:300 "/>
    <Button title={"Linear Gradient"} colors={ ["#c278ed", "#991fe0", "#5c1287",] } textStyle={[{ color: "#FFF" }]} />
    <Button rounded title={"Linear Gradient Rounded"} colors={ ["#c278ed", "#991fe0", "#5c1287",] } textStyle={[{ color: "#FFF" }]} />
  </View>
))
createButtonStory( "Button_Regular", (<Button title="Regular"/>) );
createButtonStory( "Button_Rounded", (<Button rounded title="Rounded"/>) );
createButtonStory( "Button_Full", (<Button full title="Full"/>) );
createButtonStory( "Button_Block", (<Button block title="Block"/>) );

createButtonStory( "Button_Disabled", (<Button disabled title="Diabled"/>) );
createButtonStory( "Button_Primary", (<Button primary title="Primary"/>) );
createButtonStory( "Button_Secundary", (<Button secondary title="Secondary"/>) );
createButtonStory( "Button_Info", (<Button info title="Info"/>) );
createButtonStory( "Button_Success", (<Button success title="Success"/>) );
createButtonStory( "Button_Warning", (<Button warning title="Warning"/>) );
createButtonStory( "Button_Danger", (<Button danger title="Danger"/>) );

createButtonStory( "Button_Regular_Width", (<Button containerStyle={{ width: 300 }} title="Regular, width:300 "/>) );

createButtonStory( "Button_LinearGradient", ( <Button title={"Linear Gradient"} colors={ ["#c278ed", "#991fe0", "#5c1287",] } textStyle={[{ color: "#FFF" }]} />) )
createButtonStory( "Button_LinearGradient_Rounded", ( <Button rounded title={"Linear Gradient Rounded"} colors={ ["#c278ed", "#991fe0", "#5c1287",] } textStyle={[{ color: "#FFF" }]} />) )

