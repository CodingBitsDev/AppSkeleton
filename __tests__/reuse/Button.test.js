import "react-native";
import React, {useState} from "react";

import Button from "reuse/Button.js";

import renderer from "react-test-renderer";

describe( "ComponentTest", () => {
  it("Should Render correctly", () => {
    let ButtonRender = renderer.create(<Button />).toJSON();
    expect(ButtonRender).toMatchSnapshot();
  });

  it("Should Render correctly when disabled is true", () => {
    let ButtonRender = renderer.create(<Button disabled />);
    let ButtonJSON =  ButtonRender.toJSON();
    expect(ButtonJSON).toMatchSnapshot();
  });

  it("Should Render correctly when primary is true", () => {
    let ButtonRender = renderer.create(<Button primary />);
    let ButtonJSON =  ButtonRender.toJSON();
    expect(ButtonJSON).toMatchSnapshot();
  });

  it("Should Render correctly when secondary is true", () => {
    let ButtonRender = renderer.create(<Button secondary />);
    let ButtonJSON =  ButtonRender.toJSON();
    expect(ButtonJSON).toMatchSnapshot();
  });

  it("Should Render correctly when success is true", () => {
    let ButtonRender = renderer.create(<Button success />);
    let ButtonJSON =  ButtonRender.toJSON();
    expect(ButtonJSON).toMatchSnapshot();
  });

  it("Should Render correctly when danger is true", () => {
    let ButtonRender = renderer.create(<Button danger />);
    let ButtonJSON =  ButtonRender.toJSON();
    expect(ButtonJSON).toMatchSnapshot();
  });

  it("Should Render correctly when warning is true", () => {
    let ButtonRender = renderer.create(<Button warning />);
    let ButtonJSON =  ButtonRender.toJSON();
    expect(ButtonJSON).toMatchSnapshot();
  });

  it("Should Render correctly when info is true", () => {
    let ButtonRender = renderer.create(<Button info />);
    let ButtonJSON =  ButtonRender.toJSON();
    expect(ButtonJSON).toMatchSnapshot();
  });

  it("Should Render correctly when styles are given", () => {
    let ButtonRender = renderer.create(<Button style={{ padding: 20 }} />);
    let ButtonJSON =  ButtonRender.toJSON();
    expect(ButtonJSON).toMatchSnapshot();
  });

  it("Should Render with 1 child", () => {
    let ButtonRender = renderer.create(<Button />);
    let ButtonJSON =  ButtonRender.toJSON();
    expect(ButtonJSON.children.length).toBe(1);
  });

});
