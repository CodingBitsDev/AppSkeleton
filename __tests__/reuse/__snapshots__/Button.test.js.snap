// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentTest Should Render correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 2.5,
      "elevation": 10,
      "justifyContent": "center",
      "minHeight": 30,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.34,
      "shadowRadius": 6.27,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    onLayout={[Function]}
    proxiedProperties={
      Object {
        "colors": Array [
          4294967295,
          4294967295,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "startPoint": undefined,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "justifyContent": "center",
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 16,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;

exports[`ComponentTest Should Render correctly when danger is true 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 2.5,
      "elevation": 10,
      "justifyContent": "center",
      "minHeight": 30,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.34,
      "shadowRadius": 6.27,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    onLayout={[Function]}
    proxiedProperties={
      Object {
        "colors": Array [
          4292432719,
          4292432719,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "startPoint": undefined,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "justifyContent": "center",
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;

exports[`ComponentTest Should Render correctly when disabled is true 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 2.5,
      "elevation": 10,
      "justifyContent": "center",
      "minHeight": 30,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.34,
      "shadowRadius": 6.27,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    onLayout={[Function]}
    proxiedProperties={
      Object {
        "colors": Array [
          4288256409,
          4288256409,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "startPoint": undefined,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "justifyContent": "center",
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 16,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;

exports[`ComponentTest Should Render correctly when info is true 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 2.5,
      "elevation": 10,
      "justifyContent": "center",
      "minHeight": 30,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.34,
      "shadowRadius": 6.27,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    onLayout={[Function]}
    proxiedProperties={
      Object {
        "colors": Array [
          4284658166,
          4284658166,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "startPoint": undefined,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "justifyContent": "center",
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 16,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;

exports[`ComponentTest Should Render correctly when primary is true 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 2.5,
      "elevation": 10,
      "justifyContent": "center",
      "minHeight": 30,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.34,
      "shadowRadius": 6.27,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    onLayout={[Function]}
    proxiedProperties={
      Object {
        "colors": Array [
          4290935021,
          4290935021,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "startPoint": undefined,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "justifyContent": "center",
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 16,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;

exports[`ComponentTest Should Render correctly when secondary is true 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 2.5,
      "elevation": 10,
      "justifyContent": "center",
      "minHeight": 30,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.34,
      "shadowRadius": 6.27,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    onLayout={[Function]}
    proxiedProperties={
      Object {
        "colors": Array [
          undefined,
          undefined,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "startPoint": undefined,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "justifyContent": "center",
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;

exports[`ComponentTest Should Render correctly when styles are given 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 2.5,
      "elevation": 10,
      "justifyContent": "center",
      "minHeight": 30,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.34,
      "shadowRadius": 6.27,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    onLayout={[Function]}
    proxiedProperties={
      Object {
        "colors": Array [
          4294967295,
          4294967295,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "startPoint": undefined,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "justifyContent": "center",
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 16,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;

exports[`ComponentTest Should Render correctly when success is true 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 2.5,
      "elevation": 10,
      "justifyContent": "center",
      "minHeight": 30,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.34,
      "shadowRadius": 6.27,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    onLayout={[Function]}
    proxiedProperties={
      Object {
        "colors": Array [
          4284266588,
          4284266588,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "startPoint": undefined,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "justifyContent": "center",
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 16,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;

exports[`ComponentTest Should Render correctly when warning is true 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 2.5,
      "elevation": 10,
      "justifyContent": "center",
      "minHeight": 30,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.34,
      "shadowRadius": 6.27,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    onLayout={[Function]}
    proxiedProperties={
      Object {
        "colors": Array [
          4293963086,
          4293963086,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "startPoint": undefined,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "justifyContent": "center",
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 16,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
